packages/mobile/src/components/image/TrackImage.tsx

```diff
@@ -27,14 +28,24 @@ export const useTrackImage = (
 }
 
 type TrackImageProps = {
-  track: Parameters<typeof useTrackImage>[0]
+  track: Track
   user?: Parameters<typeof useTrackImage>[1]
 } & DynamicImageProps
 
 export const TrackImage = (props: TrackImageProps) => {
   const { track, user, ...imageProps } = props
 
+  const { source: localSource, handleError: localHandleError } =
```

Comment:
this is cool. so im wondering where is the best place to introduce this local hook... would it make sense to build into the existing useTrackImage? that way we can kinda isolate the sources and it's usage is the same across the app, like "i dont care where i get it from, i just expect to get something when i call getImage?"

Reply:
Yeah I was debating which way to do it. I think that would make sense. I guess there's also a question of does album art change regularly? Should we always pull from discovery if we're online, or is it okay to default to the local cache and maybe miss an update until the offline verifier comes back around to it.

Reply:
Yeah my guess is like eager from local, make a fetch if online, and break otherwise?